sda/internal/config/config_test.go

```diff
@@ -364,6 +365,99 @@ func (suite *ConfigTestSuite) TestGetC4GHKey() {
 	defer os.RemoveAll(keyPath)
 }
 
+func (suite *ConfigTestSuite) TestGetC4GHprivateKeys_AllOK() {
+	// Generate a crypth4gh keypair
+	_, privateKey, err := keys.GenerateKeyPair()
+	assert.NoError(suite.T(), err)
+
+	keyPath, _ := os.MkdirTemp("", "key")
+	keyFile1 := keyPath + "/c4gh1.key"
+	keyFile2 := keyPath + "/c4gh2.key"
+
+	// Write the first private key file
+	keyFileWriter1, err := os.Create(keyFile1)
+	assert.NoError(suite.T(), err)
+	err = keys.WriteCrypt4GHX25519PrivateKey(keyFileWriter1, privateKey, []byte("test"))
+	assert.NoError(suite.T(), err)
+	keyFileWriter1.Close()
+
+	// Write the second private key file
+	keyFileWriter2, err := os.Create(keyFile2)
+	assert.NoError(suite.T(), err)
+	err = keys.WriteCrypt4GHX25519PrivateKey(keyFileWriter2, privateKey, []byte("test"))
+	assert.NoError(suite.T(), err)
+	keyFileWriter2.Close()
```

Comment:
Instead of doing this for every test that needs the keys and encrypted files make use of the `setupTest()` or `setupSuite()` functions. The first is run before every test and the later is run once before any tests are run.

Reply:
Fixed in the commit b81ce66c and see also the reply in https://github.com/neicnordic/sensitive-data-archive/pull/1197#discussion_r1892175222